### PR TITLE
Create exercise matching-brackets

### PR DESCRIPTION
**Pull request by PaulT89**
## Summary

Used generator to create Matching Brackets practice exercise.

Difficulty = 2.  Oddly enough, Python and Rust have this as 1, and Ruby has it at 7 - it's definitely not 7, and 1 seems a little low.

## Checklist
- [x] If docs where changed run `./bin/configlet generate` to ensure all documents are properly generated.
- [x] CI is green

### CODE REVIEW

**Review by verdammelt — COMMENTED** [review]
Two comments which are what I'd give as feedback if your example solution were a submitted solution:

The knowledge of brackets and their pairs is spread in two places, I think it would be good form to collect that together. 

Also replace the hard-coding of matching checking with some table lookup type thing perhaps. This makes the code more extensible.

**Comment by PaulT89** [comment]
Good idea.

File: exercises/practice/matching-brackets/.meta/example.lisp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+(defparameter +bracket-lookup+ '((#\( . #\)) (#\{ . #\}) (#\[ . #\])))
+
+(defun eliminate-pairs (tracker bracket)
+  (let ((match (rassoc bracket +bracket-lookup+)))
+    (if (and tracker match (char= (car match) (car tracker)))
+      (cdr tracker)
+      (cons bracket tracker))))
+
+(defun pairedp (value)
+  (let ((only-brackets (remove-if-not (lambda (c) (find c "{}[]()")) value)))
```
Comment (verdammelt) [inline exercises/practice/matching-brackets/.meta/example.lisp:16]: Still have the knowledge of brackets split up... maybe generate the the set of all brackets from `+bracket-lookup+`?

Comment (PaulT89) [reply exercises/practice/matching-brackets/.meta/example.lisp:16]: Okay.

**Pull request merged**